modules/javafx.controls/src/main/java/javafx/scene/control/skin/ToolBarSkin.java

```diff
@@ -552,17 +561,33 @@ public CssMetaData<ToolBar,Pos> getCssMetaData() {
     private void initialize() {
         if (getSkinnable().getOrientation() == Orientation.VERTICAL) {
             box = new VBox();
+            overflowBox = new VBox();
         } else {
             box = new HBox();
+            overflowBox = new HBox();
         }
         box.getStyleClass().add("container");
         box.getChildren().addAll(getSkinnable().getItems());
-        overflowMenu = new ToolBarOverflowMenu(overflowMenuItems);
+        // The overflowBox must have the same style classes, otherwise the overflow items may get wrong values.
+        overflowBox.idProperty().bind(box.idProperty());
+        Bindings.bindContent(overflowBox.getStyleClass(), box.getStyleClass());
+        Bindings.bindContent(overflowBox.getStylesheets(), box.getStylesheets());
+        box.getPseudoClassStates().addListener((SetChangeListener<? super PseudoClass>) change -> {
```

Comment:
you can also use Bindings.bindContent(Set,ObservableSet)

Reply:
line 575 still uses a listener instead of Bindings.bindContent(**Set,ObservableSet**)

Reply:
The ``Node.getPseudoClassStates()`` returns an unmodifiable collection. The bindings would not work. Therefore we need to use a listener and change the pseudo-class-state by using the ``Node.pseudoClassStateChanged(PseudoClass, boolean)`` method.

Reply:
how silly of me!  you are right, of course.  thank you for clarification.

Reply:
You are welcome, partner :-). 
It was a surprise for me too. 
It's strange, why it needs to be an unmodifiable collection.

Reply:
good question!  admittedly, it might be easier to control what happens as a result of a change if the only mutator is `pseudoClassStateChanged()` method.